src/plugins/charset_converter.c

```diff
@@ -0,0 +1,185 @@
+#include "plugins/charset_converter.h"
+#include "flutter-pi.h"
+#include "pluginregistry.h"
+#include "util/logging.h"
+#include <iconv.h>
+
+static bool convert(char *buf, char *outbuf, size_t len, const char *from, const char *to)
+{
+    iconv_t iconv_cd;
+    if ((iconv_cd = iconv_open(to, from)) == (iconv_t) -1) {
+        LOG_ERROR("Cannot open iconv from %s to %s\n", from, to);
+        return false;
+    }
+
+    char *inbuf = buf;
+    size_t inlen = len;
+    size_t outlen = len;
+    size_t res = 0;
+
+    while (inlen > 0 && outlen > 0) {
+        res = iconv(iconv_cd, &inbuf, &inlen, &outbuf, &outlen);
+        if (res == 0)
+            break;
+
+        if (res == (size_t) (-1))
+        {
+            if (errno != EILSEQ && errno != EINVAL)
+            {
+                iconv_close(iconv_cd);
+                *outbuf = '\0';
+
+                return true;
+            }
+            else if (inbuf < outbuf) {
+                iconv_close(iconv_cd);
+                *outbuf = '\0';
+
+                return true;
+            }
+        }
+
+        if (inlen > 0 && outlen > 0) {
+            *outbuf++ = *inbuf++;
+            inlen--;
+            outlen--;
+        }
+    }
+
+    iconv_close(iconv_cd);
+    *outbuf = '\0';
+    return true;
+}
+
+static int on_encode(struct platch_obj *object, FlutterPlatformMessageResponseHandle *responseHandle) {
+    struct std_value *args, *tmp;
+    char *charset, *data;
+
+    args = &object->std_arg;
+
+    if (args == NULL || !STDVALUE_IS_MAP(*args)) {
+        return platch_respond_illegal_arg_std(responseHandle, "Expected `arg` to be a map.");
+    }
+
+    tmp = stdmap_get_str(&object->std_arg, "charset");
+    if (tmp == NULL || !STDVALUE_IS_STRING(*tmp)) {
+        LOG_ERROR("Call missing mandatory parameter charset.\n");
+        return platch_respond_illegal_arg_std(responseHandle, "Expected `arg['charset'] to be a string.");
+    }
+
+    charset = STDVALUE_AS_STRING(*tmp);
+
+    tmp = stdmap_get_str(&object->std_arg, "data");
+    if (tmp == NULL || !STDVALUE_IS_STRING(*tmp)) {
+        LOG_ERROR("Call missing mandatory parameter data.\n");
+        return platch_respond_illegal_arg_std(responseHandle, "Expected `arg['data'] to be a string.");
+    }
+
+    data = STDVALUE_AS_STRING(*tmp);
+
+    char* from = (char*) malloc(strlen(data) + 1);
+    char* to = (char*) malloc(strlen(data) + 1);
+    strcpy(from, data);
+
+    bool res = convert(from, to, strlen(from) + 1, "UTF-8", charset);
+    if(!res)
+    {
+        return platch_respond_error_std(responseHandle, "error_id", "charset_name_unrecognized", NULL);
+    }
+
+    uint8_t* output;
+    output = (uint8_t*)malloc(strlen(to));
+
+    for(int i = 0; i < strlen(to); i++)
+    {
+        output[i] = (uint8_t)to[i];
+    }
```

Comment:
Why are you copying the string into a new buffer here? Can't you just use `to` directly? :)

Also avoid the explicit (uint8_t*) cast here. C++ requires that, but it's rarely done in C.